### DISCUSSION TITLE
Cannot update parent href attribute of selected component

### DISCUSSION THREAD

**Discussion by datdang2412** [GrapesJS/grapesjs#4946] [Q&A] [answered]
Hi guys, when changing current src attribute of an image using `editor?.getSelected()?.set('src', photo.path)` and it works. But I also want to change the href of the parent using `editor?.getSelected()?.parent()?.set('href', photo.path)` 

The model tells that it's changed but when getting the output from editor.getHtml() it returns old attribute value:
<img width="1597" alt="Screenshot 2023-02-27 at 06 17 30" src="https://user-images.githubusercontent.com/49665252/221443422-c6dbfd39-c10e-44e3-acfc-ffcb606ea4bb.png">

This is how the dom structure between the selected and the parent:
<img width="810" alt="Screenshot 2023-02-27 at 06 22 25" src="https://user-images.githubusercontent.com/49665252/221443598-c0a3f33e-3798-44a3-94b4-7b0310bc81b5.png">

Not sure if this is a bug, i'm using version: 0.20.1

**Comment by artf** [comment, marked as answer]
`src` is a [property of the Image component](https://github.com/GrapesJS/grapesjs/blob/45f23acf975d8ab09b60c095a3b405002b8c1658/src/dom_components/model/ComponentImage.ts#L23), the [Link doesn't have any property](https://github.com/GrapesJS/grapesjs/blob/45f23acf975d8ab09b60c095a3b405002b8c1658/src/dom_components/model/ComponentLink.ts#L12-L14), only attributes, so this is what you should do
```js
editor?.getSelected()?.parent()?.addAttributes({ 'href': photo.path });
```

**Comment by datdang2412** [comment]
Hi @artf thanks for the answer! It worked.
